### PR TITLE
[PERF] Move performance testing logic into python scripts

### PR DESCRIPTION
**Pull request by caaavik-msft**
This change is being made so that as much of the performance testing logic is runnable and testable locally as possible, and centralised inside the performance repository rather than the runtime repository. Right now, a lot of the logic sits inside Azure Pipelines yaml files that are difficult to debug and can't be run locally. With this, it will be possible to build on top of it support to schedule performance tests to be sent to Helix from a local machine using local code.

In addition, this change also moves the logic that checks out the performance repository from a script into an Azure Pipelines `checkout` step using Azure Pipeline's [multi-repo checkout functionality](https://learn.microsoft.com/en-us/azure/devops/pipelines/repos/multi-repo-checkout?view=azure-devops). With this, you can now easily see the version of the runtime and performance repository being used for a performance test run. I have also added a "branch" parameter to the runtime performance test pipelines which will let you specify a custom branch off the performance repository to be used in case you have a change that needs to be coordinated against both repositories at the same time.

This change is dependent on https://github.com/dotnet/performance/pull/4156 and must be merged after the change to the performance repository is merged in. Both PRs should be reviewed as one piece however as they are both built for the same feature.

### CODE REVIEW

**Comment by caaavik-msft** [comment]
Seems that the draft PR got automatically assigned some reviewers, I'll reping the reviewers once I move this out of draft as it's not ready for review just yet outside of my team.

File: eng/pipelines/coreclr/templates/perf-job.yml [hunk 1]
```diff
@@ -66,6 +65,7 @@ jobs:
     iosStripSymbols: ${{ parameters.iosStripSymbols }}
     hybridGlobalization: ${{ parameters.hybridGlobalization }}
     timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+    extraSetupParameters: $(extraSetupParameters)
```
Comment (LoopedBard3) [inline eng/pipelines/coreclr/templates/perf-job.yml:68]: Where is the extraSetupParameters coming from? Does the initialization below get replaced above/here.

File: src/mono/wasm/emsdk
Comment (LoopedBard3) [inline src/mono/wasm/emsdk:1]: Is this added folder(?) added on purpose?

File: eng/pipelines/coreclr/templates/perf-job.yml (hunk 1 above)
Comment (caaavik-msft) [reply eng/pipelines/coreclr/templates/perf-job.yml:68]: This variable is defined lower in the variables section, it does seem to work properly in my testing

**Review by LoopedBard3 — APPROVED** [review]
This runtime portion of the changes looks good, especially if we are able to get a mostly successful test through before the full merge.

**Review by DrewScoggins — APPROVED** [review]

**Pull request merged**